src/middlewares.js

```diff
@@ -83,9 +82,10 @@ function handleParseHeaders(req, res, next) {
   }
 
   info.app = cache.apps.get(info.appId);
-  req.config = new Config(info.appId, mount);
+  req.config = cache.apps.get(info.appId);
+  req.config.mount = mount;
```

Comment:
now cache.apps.get returns a frozen object, mount is on a getter/setter, that's the only mutable part